server/models/companyModel.js

```diff
@@ -36,4 +37,52 @@ const companySchema = new Schema({
     }
 }, { timestamps: true })
 
+//static signup method
+
+companySchema.statics.signup = async function (name, industry, location, size,email, password) {
+
+    // validation
+    if (!email || !password) {
+        throw Error('All fields must be filled')
+    }
+    if (!validator.isEmail(email)) {
+        throw Error('Email is not Valid')
+
+    }
+    if (!validator.isStrongPassword(password)) {
+        throw Error('Password is not strong enough')
+    }
+
+    const exists = await this.findOne({ email })
+
+    if (exists) {
+        throw Error('Email already in use')
+    }
+    const salt = await bcrypt.genSalt(10)
```

Comment:
you can extract the salt to a config or env as well if you want to :)

Reply:
I just put the Secret code of the encryption there. isn't enough ?